### ISSUE TITLE
[LifetimeSafety] False-negative with dangling pointer fields in base class

### ISSUE THREAD

**Issue by usx95** [llvm/llvm-project#191823]
Use after scope via a base-field
```cpp
struct X {
    int* x;
};

struct Y : X {
    int* y;
    void bar() {
        {
            int a;
            x = &a;
        }
        (void)x; // Missing use-after-scope!
    }
};
```
https://godbolt.org/z/6ds9GM4c5

Same for dangling-field, noescape escape to field, lifetime annotation suggestion in constructors.
